eslint.config.cjs

```diff
@@ -1,92 +1,91 @@
-  languageOptions: {
-    parser: tsParser,
-    parserOptions: {
-      ecmaFeatures: { modules: true },
-      ecmaVersion: '2022',
-      project: [
-        'tests/tsconfig.json'// this is the config that includes both `src` and `tests` directories
-      ]
+    // extends       : ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
+    languageOptions: {
+      parser: tsParser,
+      parserOptions: {
+        ecmaFeatures: { modules: true },
+        ecmaVersion: "2022",
+        project: [
+          "tests/tsconfig.json", // this is the config that includes both `src` and `tests` directories
+        ],
+      },
+      globals: {
+        ...globals.node,
+        ...globals.es2021,
+        ...globals.browser,
+        console: "readonly",
+      },
     },
-    globals: {
-      ...globals.node,
-      ...globals.es2021,
-      ...globals.browser
-    }
+    plugins: {
+      "@typescript-eslint": tsPlugin,
+      mocha: mochaPlugin,
+    },
+    files: ["**/*.ts"],
+    // IMPORTANT and confusing: `ignores` only exclude files from the `files` setting.
+    // To exclude *.js files entirely, you need to have a separate config object altogether. (See another `ignores` below.)
+    ignores: ["**/*.d.ts"],
+    rules: {
+      "key-spacing": [
+        "error",
+        {
+          align: {
+            afterColon: true,
+            beforeColon: true,
+            on: "colon",
+          },
+        },
+      ],
+      quotes: ["error", "single", { allowTemplateLiterals: true }],
+      semi: ["error", "always"],
+      indent: ["error", 2, { SwitchCase: 1 }],
+      "no-unused-vars": "off",
+      "prefer-const": "off",
+      "@typescript-eslint/no-unused-vars": [
+        "error",
+        {
+          vars: "all",
+          args: "after-used",
+          ignoreRestSiblings: true,
+          argsIgnorePattern: "^_",
+          varsIgnorePattern: "^_",
+        },
+      ],
+      "no-dupe-class-members": "off",
+      "no-trailing-spaces": ["error"],
+      "@typescript-eslint/no-explicit-any": "off",
+      "@typescript-eslint/no-non-null-assertion": "off",
+      "@typescript-eslint/ban-ts-comment": "off",
+      "@typescript-eslint/no-unused-vars": "off",
+      // TODO: Revisit new default mocha rules that were disabled in #579 - https://github.com/TBD54566975/web5-js/issues/580
+      "mocha/no-exclusive-tests": "warn",
+      "mocha/no-setup-in-describe": "off",
+      "mocha/no-mocha-arrows": "off",
+      "mocha/max-top-level-suites": "off",
+      "mocha/no-identical-title": "off",
+      "mocha/no-pending-tests": "off",
+      "mocha/no-skipped-tests": "off",
+      "mocha/no-sibling-hooks": "off",
+    },
+  },
+  {
+    ignores: ["**/*.js", "**/*.cjs", "**/*.mjs"],
   },
-  plugins: {
-    '@typescript-eslint': tsPlugin,
-    'mocha': mochaPlugin
+  {
+    files: ["**/*.ts"],
+    rules: {
+      "no-undef": "off",
+      "no-redeclare": "off",
```

Comment:
The Ts compiler handles these errors correctly while eslint is dumber and throwing when it shouldn't. When I looked into the situation it turned out that you're not supposed to have these rules enabled for TS due to overlap with the compiler

Reply:
I see... though it appears this updated eslint config actually causes more error (see CICD log failing immediately in `common` pacakge), I tried with the original config, no errors. You'll need to investigate, please leave inline comment if this turns out to be necessary so that this valuable tribal context knowledge is not lost.

Reply:
will do. i'll look into the extra errors but i doubt they're from turning rules **off** they're probably from something else i did

Reply:
I fixed the errors here. Keep in mind the eslint recommended config is not for Typescript but for javascript:
<img width="752" alt="Screenshot 2024-08-05 at 4 14 23 PM" src="https://github.com/user-attachments/assets/ff7c20a8-84ee-40d7-a657-e858e22b6a04">

That being said I don't want to spend much more time on this since I expect we'll probably move to Biome or a better option than bloated eslint